pkg/strategy/grid2/strategy.go

```diff
@@ -1385,9 +1385,36 @@ func (s *Strategy) recoverGridWithOpenOrdersByScanningTrades(ctx context.Context
 	}
 
 	// 1. build pin-order map
-	// 2. fill the pin-order map by querying trades
+	pinOrderMapWithOpenOrders, err := s.buildPinOrderMap(grid, openOrdersOnGrid)
+	if err != nil {
+		return errors.Wrapf(err, "failed to build pin order map with open orders")
+	}
+
+	// 2. build the filled pin-order map by querying trades
+	pinOrderMapWithFilledOrders, err := s.buildFilledPinOrderMapFromTrades(ctx, historyService, pinOrderMapWithOpenOrders)
```

Comment:
you can simplify the var name, like `pinOrders`

Reply:
I will keep it first to know it's only for filled orders.

Reply:
then you can use `pinOrdersFilled` and `pinOrdersOpen`